code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -155,9 +159,90 @@
+	use_state_flags = XACT_USE_BUSY
+
+/datum/action/xeno_action/select_reagent/update_button_icon()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/atom/A = X.selected_reagent
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/actions.dmi', button, initial(A.name))
+	return ..()
+
+/datum/action/xeno_action/select_reagent/action_activate()
+	var/mob/living/carbon/xenomorph/X = owner
+	var/list/available_reagents = X.xeno_caste.available_reagents_define
+	var/i = available_reagents.Find(X.selected_reagent)
+	if(length(available_reagents) == i)
+		X.selected_reagent = available_reagents[1]
+	else
+		X.selected_reagent = available_reagents[i+1]
+	var/atom/A = X.selected_reagent
+	to_chat(X, "<span class='notice'>We will now slash with <b>[initial(A.name)]</b>.</span>")
+	update_button_icon()
+	return succeed_activate()
+
+// ***************************************
+// *********** Reagent slash
+// ***************************************
+/datum/action/xeno_action/activable/reagent_slash
+	name = "Reagent Slash"
+	mechanics_text = "Deals damage 4 times and injects 3u of selected reagent per slash. Can move while slashing."
+	ability_name = "reagent slash"
+	cooldown_timer = 6 SECONDS
+	plasma_cost = 100
+	target_flags = XABB_MOB_TARGET
+
+/datum/action/xeno_action/activable/reagent_slash/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+	if(get_dist(owner, A) > 1)
+		to_chat(owner, "<span class='warning'>We need to be next to our prey.</span>")
+		return FALSE
+	if(!A.can_sting())
+		to_chat(owner, "<span class='warning'>Our slash won't affect this target!</span>")
+		return FALSE
+	return TRUE
+
+/datum/action/xeno_action/activable/reagent_slash/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
+	succeed_activate()
+	slash_action(A, X.selected_reagent, channel_time = DEFILER_REAGENT_SLASH_DELAY, count = DEFILER_REAGENT_SLASH_COUNT, transfer_amount = DEFILER_REAGENT_SLASH_U_AMOUNT)
+	add_cooldown()
+
+/datum/action/xeno_action/activable/reagent_slash/proc/slash_action(mob/living/carbon/C, toxin = /datum/reagent/toxin/xeno_neurotoxin, channel_time = 1 SECONDS, transfer_amount = 4, count = 3)
+	var/mob/living/carbon/xenomorph/X = owner
+	var/datum/limb/affecting = X.zone_selected
+	var/datum/reagent/body_tox
+	var/i = 1
+	do
+		if(get_dist(X, C) > 1)
+			to_chat(X, "<span class='warning'>We need to be closer to [C].</span>")
+			return
+		if(X.stagger)
+			return 
+		X.face_atom(C)
+		body_tox = C.reagents.get_reagent(toxin)
+		C.reagents.add_reagent(toxin, transfer_amount)
+		if(!body_tox) //Let's check this each time because depending on the metabolization rate it can disappear between slashes.
+			body_tox = C.reagents.get_reagent(toxin)
```

Comment:
Why is body_tox needed here?

Reply:
removed body_tox as it seems like it was used for stopping recurring injection for things that don't apply to reagent slash